app/models/competition.rb

```diff
@@ -1195,6 +1195,15 @@ def registration_edits_currently_permitted?
     errors.add(:event_change_deadline_date, I18n.t('competitions.errors.event_change_deadline_after_end_date')) if event_change_deadline_date > end_date.to_datetime.end_of_day
   end
 
+  def enabling_on_the_spot_registration?
+    self.on_the_spot_registration_changed? && self.on_the_spot_registration?
+  end
+
+  validate :on_the_spot_registration_must_be_valid
```

Comment:
I think we can get a better name here - what we're really checking is the consistency of the ots date with the event change deadline, right? Perhaps `enforce_edit_deadline_ots_consistency` or something similar? 

(I'm not a huge fan of acronyms in variable names but the tradeoff here feels worth it)

Reply:
I was trying to be consistent with the other validation names and patterns directly above, but to be fair I'm not a big fan of those names either.

Would you rather sacrifice consistency for a better name here locally, or sacrifice the PR scope by renaming the other methods as well?

Reply:
I'm all for sacrificing consistency, and will note down the renames for a potential future/downtime PR